Web Quiz Engine (Kotlin)/task/src/engine/WebQuizApiController.kt

```diff
@@ -1,69 +1,46 @@
 package engine
 
+import engine.entity.*
+import engine.service.QuizService
 import jakarta.validation.Valid
-import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.http.HttpStatus
 import org.springframework.http.ResponseEntity
 import org.springframework.web.bind.annotation.*
 
-
-// exception dans leur propre package
-// pas de logique dans POJO
-// chaque POJO dans son ficher dans un package model ?
-// inclue repsonse et request body
-// pas de logique dans le controller
-
-
-data class AnswerQuizRequestBody(val answer: List<Int>?)
 @RestController
-class WebQuizApiController {
-
-    @Autowired
-    private lateinit var quizRepository: WebQuizRepository
-
-    sealed class QuizPostResponseBody(val success: Boolean, val feedback: String)
-    class QuizPostResponseBodyCorrect(success: Boolean = true, feedback: String = "Congratulations, you're right!")
-        : QuizPostResponseBody(success, feedback)
-    class QuizPostResponseBodyIncorrect(success: Boolean = false, feedback: String = "Wrong answer! Please, try again.")
-        : QuizPostResponseBody(success, feedback)
+class WebQuizApiController(private val quizService: QuizService) {
 
     @PostMapping("/api/quizzes/{quizId}/solve")
     fun answerQuiz(@RequestBody reqBody: AnswerQuizRequestBody, @PathVariable quizId: Int):
             ResponseEntity<QuizPostResponseBody> =
-        quizRepository.getQuizById(id = quizId)?.let { quiz ->
-            ResponseEntity(
-                if ((quiz.answerList.isNullOrEmpty() && reqBody.answer.isNullOrEmpty()) ||
-                    (quiz.answerList?.containsAll(reqBody.answer.orEmpty()) == true &&
-                            reqBody.answer?.containsAll(quiz.answerList) == true))
-                    QuizPostResponseBodyCorrect()
-                else
-                    QuizPostResponseBodyIncorrect(),
-                HttpStatus.OK,
-            )
-        } ?: throw QuizNotFoundException(quizId)
+        if (quizService.isQuizSolution(
+                quizId = quizId,
+                answerList = reqBody.answerList)) {
+            ResponseEntity(QuizPostResponseBodyCorrect(), HttpStatus.OK)
+        } else {
+            ResponseEntity(QuizPostResponseBodyIncorrect(), HttpStatus.OK)
```

Comment:
I know there were some trick to not have to send the `OK` status each time but unfortunately it escaped my mind.

Reply:
See my comment above on the method signature.

This method should look like this in the end:
```
@PostMapping("/api/quizzes/{quizId}/solve")
@ResposeStatus(HttpStatus.OK)
fun answerQuiz(@RequestBody reqBody: AnswerQuizRequestBody, @PathVariable quizId: Int): QuizPostResponseBody =
    quizService.isQuizSolution(quizId = quizId, answerList = reqBody.answerList)
```

Reply:
oh nice, this is so concise now! Done